src/core/core/task.cpp

```diff
@@ -465,16 +465,13 @@ timer_task::timer_task(
 
 void timer_task::exec()
 {
-    task_state RUNNING_STATE = TASK_STATE_RUNNING;
-    
     _cb(_context);
 
     if (_interval_milliseconds > 0)
     {
-        if (_state.compare_exchange_strong(RUNNING_STATE, TASK_STATE_READY))
-        {
-            set_delay(_interval_milliseconds);            
-        }        
+        //_state must be TASK_STATE_RUNNING here
```

Comment:
this is true by design. let's have a dbg_assert though here to detect implementation bugs.